### ISSUE TITLE
ejabberd_tls uses undefined function

### ISSUE THREAD

**Issue by arcusfelis** [esl/MongooseIM#642]
```erlang
-ifdef(SSL40).                                                                       
-define(CERT_DECODE, {public_key, pkix_decode_cert, plain}).                         
-define(CERT_SELFSIGNED, {public_key, pkix_is_self_signed}).                         
-else.                                                                               
-define(CERT_DECODE, {ssl_pkix, decode_cert, [pkix]}).                               
-define(CERT_SELFSIGNED, {erlang, is_atom}). %% Dummy function for old OTPs          
-endif. 
```

From ejabberd v2.1.10:

```erlang
   OTP = (catch erlang:system_info(otp_release)),                                    
   if                                                                                
    OTP >= "R14" -> "-DSSL40\n";                                                     
    OTP >= "R12" -> "-DSSL39\n";                                                     
        true -> "\n"                                                                 
   end. 
```

So, SSL40 is not defined but should be. And we call an undefined module ssl_pkix.

Related https://github.com/esl/MongooseIM/pull/636

**Comment by arcusfelis** [comment]
After fixing this test became "negative" http://arcusfelis.github.io/mongooseim-ct-reports/1194.1/ct_run.test@testing-gce-3e27829f-4ad0-4e0c-a07a-da9ce24084a3.2016-01-19_12.18.08/ejabberd_tests.tests.s2s_SUITE.logs/run.2016-01-19_12.34.34/s2s_suite.simple_message.206895.570.html#end

**Comment by arcusfelis** [comment]
Fixed with #636
